rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -218,84 +238,83 @@ def required_packages(cls) -> List[Text]:
 
     # training data helpers:
     @staticmethod
-    def _create_label_id_dict(training_data: "TrainingData") -> Dict[Text, int]:
+    def _create_label_id_dict(
+        training_data: "TrainingData", attribute: Text
+    ) -> Dict[Text, int]:
         """Create label_id dictionary"""
 
         distinct_label_ids = set(
-            [example.get("intent") for example in training_data.intent_examples]
-        )
+            [example.get(attribute) for example in training_data.intent_examples]
+        ) - {None}
         return {
             label_id: idx for idx, label_id in enumerate(sorted(distinct_label_ids))
         }
 
     @staticmethod
-    def _create_label_id_token_dict(
-        label_ids: List[Text], label_split_symbol: Text
-    ) -> Dict[Text, int]:
-        """Create label_id token dictionary"""
-
-        distinct_tokens = set(
-            [
-                token
-                for label_id in label_ids
-                for token in label_id.split(label_split_symbol)
-            ]
-        )
-        return {token: idx for idx, token in enumerate(sorted(distinct_tokens))}
-
-    def _create_encoded_label_ids(self, label_id_dict: Dict[Text, int]) -> np.ndarray:
-        """Create matrix with label_ids encoded in rows as bag of words.
+    def _find_example_for_label(label, examples, attribute):
+        for ex in examples:
+            if ex.get(attribute) == label:
+                return ex
+        return None
 
-        If label_tokenization_flag is off, returns identity matrix.
-        """
-
-        if self.label_tokenization_flag:
-            label_id_token_dict = self._create_label_id_token_dict(
-                list(label_id_dict.keys()), self.label_split_symbol
-            )
-
-            encoded_all_label_ids = np.zeros(
-                (len(label_id_dict), len(label_id_token_dict))
+    def _create_encoded_label_ids(
+        self,
+        training_data: "TrainingData",
+        label_id_dict: Dict[Text, int],
+        attribute: Text,
+        attribute_feature_name: Text,
+    ) -> np.ndarray:
+        """Create matrix with label_ids encoded in rows as bag of words. The features are already computed.
+        Find a training example for each label and get the encoded features from the corresponding Message object"""
+
+        encoded_all_labels = []
+
+        for label_name, idx in label_id_dict.items():
+            label_example = self._find_example_for_label(
+                label_name, training_data.intent_examples, attribute
             )
-            for key, idx in label_id_dict.items():
-                for t in key.split(self.label_split_symbol):
-                    encoded_all_label_ids[idx, label_id_token_dict[t]] = 1
+            if label_example:
+                encoded_all_labels.insert(
+                    idx, label_example.get(attribute_feature_name)
```

Comment:
I don't think `insert` works as expected in empty list. Is there a possibility that `label_id_dict` is not ordered by index?

Reply:
Can't remove label_id_dict as discussed earlier because it is used to create session data. Created a list of tuples, sorted them by index and then pulled out the ordered encodings. Thanks for catching this.

Reply:
where is it used to create session data?

Reply:
It's passed to `_create_session_data()` function because we need the label ids from actual labels.